File: src/main/java/me/testaccount666/serversystem/clickablesigns/executables/time/ActionTimeSign.java
package me.testaccount666.serversystem.clickablesigns.executables.time;

import me.testaccount666.serversystem.clickablesigns.AbstractSignClickAction;
import me.testaccount666.serversystem.userdata.User;
import me.testaccount666.serversystem.utils.ChatColor;
import org.bukkit.block.Sign;
import org.bukkit.configuration.file.FileConfiguration;

import static me.testaccount666.serversystem.utils.MessageBuilder.sign;

public class ActionTimeSign extends AbstractSignClickAction {

    @Override
    public String getBasePermissionNode() {
        return "ClickableSigns.Time";
    }

    @Override
    protected boolean executeAction(User user, Sign sign, FileConfiguration config) {
        var timeType = config.getString("TimeType", sign.getLine(1)).toLowerCase();
        timeType = ChatColor.stripColor(timeType);
        if (timeType.isEmpty()) {
            sign("Time.NoTimeSpecified", user).build();
            return false;
        }

        var world = user.getPlayer().getWorld();
        long time;

        var finalTimeType = timeType;
        switch (timeType) {
            case "day" -> time = 1000;
            case "noon" -> time = 6000;
            case "night" -> time = 13000;
            case "midnight" -> time = 18000;
            default -> {
                try {
                    time = Long.parseLong(timeType);
                } catch (NumberFormatException ignored) {
                    sign("Time.InvalidTime", user)
                            .postModifier(message -> message.replace("<TIME>", finalTimeType)).build();
                    return false;
                }
            }
        }

        world.setTime(time);
        sign("Time.TimeSet", user)
                .postModifier(message -> message.replace("<TIME>", finalTimeType)).build();
        return true;
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/clickablesigns/executables/time/ActionTimeSign.kt
package me.testaccount666.serversystem.clickablesigns.executables.time

import me.testaccount666.paperktx.colors.ChatColor.Companion.stripColor
import me.testaccount666.serversystem.clickablesigns.AbstractSignClickAction
import me.testaccount666.serversystem.extensions.signMsg
import me.testaccount666.serversystem.userdata.User
import org.bukkit.block.Sign
import org.bukkit.configuration.file.FileConfiguration

class ActionTimeSign : AbstractSignClickAction() {
    override val basePermissionNode = "ClickableSigns.Time"

    override fun executeAction(user: User, sign: Sign, config: FileConfiguration, onSuccess: () -> Unit): Boolean {
        var timeType = config.getString("TimeType", sign.getLine(1))!!.lowercase()
        timeType = stripColor(timeType)
        if (timeType.isEmpty()) {
            user.signMsg("Time.NoTimeSpecified")
            return false
        }

        val world = user.getPlayer()!!.world

        val time = when (timeType) {
            "day" -> 1000L
            "noon" -> 6000L
            "night" -> 13000L
            "midnight" -> 18000L
            else -> {
                try {
                    timeType.toLong()
                } catch (_: NumberFormatException) {
                    user.signMsg("Time.InvalidTime") {
                        postModifier { it.replace("<TIME>", timeType) }
                    }
                    return false
                }
            }
        }

        world.time = time
        user.signMsg("Time.TimeSet") {
            postModifier { it.replace("<TIME>", timeType) }
        }
        onSuccess()
        return true
    }
}